internal/testing/setup/setup.go

```diff
@@ -144,22 +136,22 @@ func NewAWSSession(ctx context.Context, t *testing.T, region string) (sess *sess
 // Otherwise, the session reads a replay file and runs the test as a replay,
 // which never makes an outgoing HTTP call and uses fake credentials.
 func NewGCPClient(ctx context.Context, t *testing.T) (client *gcp.HTTPClient, rt http.RoundTripper, done func()) {
-	var co option.ClientOption
+	c, cleanup, _ := NewRecordReplayClient(ctx, t, func(r *httpreplay.Recorder) {
+		r.ClearQueryParams("Expires")
+		r.ClearQueryParams("Signature")
+		r.ClearHeaders("Expires")
+		r.ClearHeaders("Signature")
+	})
 	if *Record {
 		creds, err := gcp.DefaultCredentials(ctx)
 		if err != nil {
 			t.Fatalf("failed to get default credentials: %v", err)
 		}
-		co = option.WithTokenSource(gcp.CredentialsTokenSource(creds))
-	} else {
-		co = option.WithoutAuthentication()
+		c, err = hrgoog.RecordClient(ctx, c, option.WithTokenSource(gcp.CredentialsTokenSource(creds)))
```

Comment:
In the go-replayers repo, I split the Google-specific parts of client configuration (the stuff involving the `api/option` package) into a separate package.